pkg/controller/machinepool/machinepool_controller.go

```diff
@@ -597,6 +597,13 @@ func (r *ReconcileMachinePool) syncMachineSets(
 					objectModified = true
 				}
 
+				// Platform updates will be blocked by webhook, unless they're not.
```

Comment:
ntr: I am *not* double-checking the override annotation here. Open for discussion, but I think it's better to leave that decision in one place. If we got past the webhook (for any reason -- even the usual hack of deleting the VWHC) we'll effect the change.

Reply:
This scares me because it opens us up to any diff that would be generated by the installer's generation where we haven't been considering drift at all, now we will blast out any changes such as with a default on a big scale.

Reply:
...which would mean
- short term the MachineSets and Machines are out of sync
- longer term if machines get replaced for whatever reason, they'll be different for hard-to-understand reasons
- ditto new machines when scaling up

And all of this of course only if hive has done a revendor since the Machines were created *and* the revendor pulled in some changed default from the installer code.

So really, really low probability... but ugly enough to warrant paranoia, and certainly outbalances doubling the check, which is easy.

On it.

Reply:
@abutcher done